src/include/OpenImageIO/imagebuf.h

```diff
@@ -1408,13 +1401,12 @@ class OIIO_API ImageBuf {
 
 
 protected:
-    std::unique_ptr<ImageBufImplBase> m_impl;  //< PIMPL idiom
+    // PIMPL idiom
+    static void impl_deleter(ImageBufImpl*);
+    std::unique_ptr<ImageBufImpl, decltype(&impl_deleter)> m_impl;
 
-    ImageBufImpl* impl() { return (ImageBufImpl*)m_impl.get(); }
-    const ImageBufImpl* impl() const
-    {
-        return (const ImageBufImpl*)m_impl.get();
-    }
+    ImageBufImpl* impl() { return m_impl.get(); }
```

Comment:
Does this help anything? Anything that can access `impl()` can also access `m_impl` which is the same length.

I doesn't look like `impl()` is used too many times in the code either ...

Reply:
Oh, you might be right. I'll take a look and eliminate it if it is only in a few places.

Reply:
It was a LOT of places. I just pushed an update, see if you are comfortable with that much change.

Reply:
I like it, I think this will be cleaner in the long run.